src/main/scala/rules/MoreCompleteExhaleSupporter.scala

```diff
@@ -281,35 +281,39 @@ object moreCompleteExhaleSupporter extends SymbolicExecutionRules {
 
         val s0 = s.copy(functionRecorder = currentFunctionRecorder)
 
-        summarise(s0, relevantChunks.toSeq, resource, args, v)((s1, snap, _, _, v1) =>
+        summarise(s0, relevantChunks.toSeq, resource, args, v)((s1, snap, _, _, v1) => {
+          val condSnap = if (v1.decider.check(Greater(perms, NoPerm()), Verifier.config.checkTimeout())) {
+            snap
+          } else {
+            Ite(Greater(perms, NoPerm()), snap.convert(sorts.Snap), Unit)
```

Comment:
Use one of those optimising term constructors (`IsPositive` etc.) here, instead of `Greater(perms, NoPerm())`?